src/main/java/converters/ArrayConverter.java

```diff
@@ -0,0 +1,322 @@
+        String[] values = input.split(",");
+
+        StringBuilder builder = new StringBuilder();
+
+        for (String value : values) {
+            builder.append(convertFromValue(value));
+        }
+
+        return builder.toString();
+    }
+
+    private static String convertFromValue(String value) throws ConverterException {
+        Format format = Format.getFormatFromInputValue(value);
+        if (format == null) {
+            throw new ConverterException(String.format("Invalid value: %s", value));
+        }
+        if (ARRAY.equals(format)){
+            return "";
+        }
+
+        Pattern regex = Pattern.compile(format.getArrayRegex());
+        Matcher matcher = regex.matcher(value);
+
+        if (matcher.find()) {
+            String valueToConvert;
+
+            if (BYTES.equals(format)){
+                if (matcher.group(2) != null) {
+                    valueToConvert = matcher.group(2);
+                } else {
+                    valueToConvert = matcher.group(1);
+                    String hexValue = "\\u00" + valueToConvert;
+                    valueToConvert = String.format("%c", Integer.parseInt(hexValue.substring(2), HEXADECIMAL));
+                }
+            } else {
+                valueToConvert = matcher.group(matcher.groupCount());
+            }
+
+            return format.getConverter().convertFrom(valueToConvert, null);
+        }
+        return value;
+    }
+
+    private void validateArrayInput(String input) throws ConverterException {
+        if (Objects.isNull(input) || input.length() < 2) {
+            throw new ConverterException(String.format("Invalid input: %s", input));
+        }
+        checkBrackets(String.valueOf(input.charAt(0)), String.valueOf(input.charAt(input.length() - 1)));
+    }
+
+    private void checkBrackets(String openingBracket, String closingBracket) throws ConverterException {
+        String closures = openingBracket + closingBracket;
+        List<String> correctBracketsOptions = List.of(CURLY_BRACKETS.getText(),
+                SQUARE_BRACKETS.getText(),
+                REGULAR_BRACKETS.getText());
+        if (!correctBracketsOptions.contains(closures)) {
+            throw new ConverterException(String.format("Non-equal brackets: %s", closingBracket));
+        }
+    }
+
+
+    private Pair convertArrayToArray(String input, int inputActualPosition, String openBrackets,
+                                     String closingBracket, IOption[] options) throws ConverterException {
+        int index = inputActualPosition;
+        List<String> values = new ArrayList<>();
+
+        StringBuilder valueToConvert = new StringBuilder();
+        String stringValue;
+        while (index < input.length()) {
+            String character = String.valueOf(input.charAt(index));
+            if (character.equals(openBrackets)) {
+                Pair semiValues = convertArrayToArray(input, index + 1, openBrackets, closingBracket, options);
```

Comment:
**issue:** use of recursion. 

Recursion is always a red flag in secure programming, as it can easily result in a stack overflow.

Reply:
**note:** stack overflow occurs at reasonable depth

using input which matches `(\{{6000}0x42\}{6000})` results in stack overflow using the default stack size `(-Xss512k)`